### PR TITLE
Add date processor migration

### PR DESCRIPTION
**Pull request by suyograo**
@original-brownbear I decided to refactor some of the names to make it more generic (and not grok specific), but most logic is the same. There is an issue with indentation with multiple hash keys, but I'm super tired and will look tomorrow :) Let me know what you think

### CODE REVIEW

File: tools/ingest-converter/src/test/java/org/logstash/ingest/DateTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public final class DateTest extends IngestTest {
+
+    @Rule
+    public final TemporaryFolder temp = new TemporaryFolder();
```
Comment (suyograo) [inline tools/ingest-converter/src/test/java/org/logstash/ingest/DateTest.java]: @original-brownbear how do I push this to the base class?

File: tools/ingest-converter/src/test/resources/org/logstash/ingest/logstashDateExtraFields.conf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+filter {
+   date {
+      match => [
+         "initial_date",
+         "dd/MM/yyyy hh:mm:ss",
+         "dd/MM/yyyy"
+      ]
+      target => "timestamp"
+timezone => "Europe/Amsterdam"
```
Comment (suyograo) [inline tools/ingest-converter/src/test/resources/org/logstash/ingest/logstashDateExtraFields.conf]: need to fix this indent

**Comment by original-brownbear** [comment]
@suyograo sorry for the delay here, not feeling so well tonight and traveling tomorrow, will get back to this in ~36h.

File: tools/ingest-converter/src/test/java/org/logstash/ingest/DateTest.java (hunk 1 above)
Comment (original-brownbear) [inline tools/ingest-converter/src/test/java/org/logstash/ingest/DateTest.java]: @suyograo you have to make the parent `IngestTest` public :)

Comment (original-brownbear) [inline tools/ingest-converter/src/test/java/org/logstash/ingest/DateTest.java]: @suyograo I assume you ran into `IllegalAccessException` here and are therefore asking?

**Comment by original-brownbear** [comment]
@suyograo fixed the moving of the `temp` folder to the parent class and the indents in https://github.com/original-brownbear/logstash/commit/819cc024dd6117836c050e2c9f65af48fe8d8197 , feel free to cherry-pick and merge if you agree :)

Also fixed missing `static`s :)

**Comment by original-brownbear** [comment]
@suyograo next step here should probably be a "dry-up", I'm happy to do that tonight once we merged this :)

**Comment by suyograo** [comment]
@original-brownbear thanks for those patches. They made sense to me, I pulled it into this one. Let me know how it looks now.

**Comment by suyograo** [comment]
> next step here should probably be a "dry-up", I'm happy to do that tonight once we merged this :)

Agreed.

**Comment by original-brownbear** [comment]
@suyograo LGTM, retriggered IT builds (those were failing for some reason, but I don't see how this PR could be the cause )

**Pull request merged**